packages/core/upload/admin/src/utils/getBreadcrumbDataCM.ts

```diff
@@ -0,0 +1,53 @@
+import { getTrad } from './getTrad';
+
+import type { Folder } from '../../../shared/contracts/folders';
+import type { MessageDescriptor } from 'react-intl';
+
+interface BreadcrumbDataFolder extends Omit<Folder, 'children' | 'files' | 'parent'> {
+  parent?: BreadcrumbDataFolder;
+  children?: {
+    count: number;
+  };
+  files?: {
+    count: number;
+  };
+}
+
+interface BreadcrumbItem {
+  id?: number | null;
+  label?: MessageDescriptor | string;
+  path?: string;
+}
+
+type BreadcrumbData = BreadcrumbItem | [];
```

Comment:
this means you can have a 2d array, as below you have an array of BreadcrumbData. Are you sure this is what you want?